File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples/CSharp/WPFListBox/WPFComponents/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Xml.Linq;
using System.Text;
using System.IO;

namespace WPFComponents
{
  public class Product
  {
    #region Public Properties
    public int ProductId { get; set; }
    public string ProductName { get; set; }
    public decimal Price { get; set; }
    public string Logo { get; set; }
    #endregion

    public IEnumerable<Product> GetProductCollection()
    {
      IEnumerable<Product> ret = null;
      string fileName;

      fileName = WPFCommon.GetCurrentDirectory() +
        @"\Xml\Product.xml";

      if (File.Exists(fileName))
        ret = Product.GetProducts(fileName);

      return ret;
    }

    public static IEnumerable<Product> GetProducts(string fileName)
    {
      XElement elem = XElement.Load(fileName);

      var items = from prod in elem.Descendants("Product")
                  select new Product
                  {
                    ProductId = Convert.ToInt32(prod.Element("ProductId").Value),
                    ProductName = prod.Element("ProductName").Value,
                    Price = Convert.ToDecimal(prod.Element("Price").Value),
                    Logo = prod.Element("Logo").Value
                  };

      return items;
    }
  }
}

File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples/VB/WPFListBox-VB/WPFComponents/Product.vb
Imports System.IO

Public Class Product
#Region "Private Variables"
  Private mProductId As Integer
  Private mProductName As String
  Private mPrice As Decimal
  Private mLogo As String
#End Region

#Region "Public Properties"
  Public Property ProductId() As Integer
    Get
      Return mProductId
    End Get
    Set(ByVal value As Integer)
      mProductId = value
    End Set
  End Property

  Public Property ProductName() As String
    Get
      Return mProductName
    End Get
    Set(ByVal value As String)
      mProductName = value
    End Set
  End Property

  Public Property Price() As Decimal
    Get
      Return mPrice
    End Get
    Set(ByVal value As Decimal)
      mPrice = value
    End Set
  End Property

  Public Property Logo() As String
    Get
      Return mLogo
    End Get
    Set(ByVal value As String)
      mLogo = value
    End Set
  End Property
#End Region

  Public Function GetProductCollection() As IEnumerable(Of Product)
    Dim ret As IEnumerable(Of Product) = Nothing
    Dim fileName As String

    fileName = WPFCommon.GetCurrentDirectory() & "\Xml\Product.xml"

    If File.Exists(fileName) Then
      ret = Product.GetProducts(fileName)
    End If

    Return ret
  End Function

  Public Shared Function GetProducts(ByVal fileName As String) As IEnumerable(Of Product)
    Dim elem As XElement = XElement.Load(fileName)

    Dim items = From prod In elem.Descendants("Product") _
        Select New Product() With { _
                    .ProductId = Convert.ToInt32(prod.Element("ProductId").Value), _
                    .ProductName = prod.Element("ProductName").Value, _
                    .Price = Convert.ToDecimal(prod.Element("Price").Value), _
                    .Logo = prod.Element("Logo").Value _
                  }

    Return items
  End Function
End Class